packer-plugin-sdk/multistep/commonsteps/step_http_server.go

```diff
@@ -54,14 +54,20 @@ func (s *StepHTTPServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Starting HTTP server on port %d", s.l.Port))
+	httpAddr = state.Get("http_ip").(string)
```

Comment:
I think this will crash with a nil pointer dereference if http_ip hasn't been set in state.

Reply:
You can use state.GetOK to protect against this: 

httpAddr, present := state.GetOk("http_ip")
if present {
... 
}